tests/dispatch/tests.py

```diff
@@ -96,6 +96,15 @@ def test_garbage_collected(self):
         self.assertEqual(result, [])
         self.assertTestIsClean(a_signal)
 
+    def test_send_robust_logger_exception(self):
+        def fails(val, **kwargs):
+            raise ValueError()
+        b_signal.connect(fails)
```

Comment:
why not use `a_signal` in this test? The signal objects seem to be identical.

Reply:
these are identical but when the test fails then a_signal doesnot disconnects which cause failures of more signals.
So b_signal is used so that only one test case in failure.

Reply:
imo these test cases should reworked to use a `try/finally` pattern to ensure the receivers are always disconnected:

```python
a_signal.connect(fails)
try:
    # the test
finally:
    a_signal.disconnect(fails)
```

Alternatively maybe they could construct the signal instances internally to the test methods.

Reply:
I think try/finally pattern is better as it will reuse the same signal so computation will be faster.
i will implement it.